application/server/main/tasks.py

```diff
@@ -33,7 +30,7 @@
 logger_console = get_logger(__name__, level=LOGGER_LEVEL)
 
 
-def create_task_harvest_partition(source_metadata_file, partition_index, total_partition_number, doi_list):
+def create_task_harvest_partition(source_metadata_file, partition_index, total_partition_number, doi_list, wiley_client):
```

Comment:
Just a question, what if we need to create a session for another client like Elsevier ? Will we need to set the client to the task too ?

Reply:
yes, we need to pass all the clients to the task if we have the session problem. Otherwise, it can be initialised when it needs to be used

Reply:
from my knowledge, this is not possible in python.
I don't see any other simple way of doing this and making sure that it is initialised only one time other than passing it as a function argument